feat: add `ndarray/base/some-by`

headlessNode opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

Progresses #2656 .

## Description

> What is the purpose of this pull request?

This pull request:

-   add `ndarray/base/some-by`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2656  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
feat: add some-by

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/main.js:
```diff
@@ -0,0 +1,176 @@
+*
+* // Create the input ndarray-like object:
+* var x = {
+*     'ref': null,
```
This should be removed in this example.

headlessNode replied:
The `ref` property?

headlessNode replied:
Yeah `ref` makes sense here that is we will be getting that from our reduction function, correct?

P.S. As a standalone function it does make sense to have it here though

kgryte replied:
Yes, the `ref` property, and, no, it should not be here. Simply because our higher order machinery, such as unary-reduce-subarray-by, passes down an object which happens to have a `ref` property, that does **not** mean it is a requisite for the function. In fact, see the README L87 which you've already added to this PR.

In our internal examples, yes, there should be a `ref` property, as we explicitly call `ndarraylike2object` within `main.js`, thus normalizing any and all ndarray-like input values such that they do have a `ref` property. But again, this is not a prerequisite of ndarrays coming into the function. E.g., I should be able to pass an `ndarray` instance, such as `ndarray/base/ctor` or `ndarray/ctor`, to this function, and it should just work. Neither of those instances have `ref` properties. So just think a bit about what our contract with the external world (i.e., outside of this API) is.

kgryte replied:
Also, read L76 in this file where we describe the minimum set of required properties.

headlessNode committed:
feat: add 4d kernels

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
feat: add ts defs

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/README.md:
```diff
@@ -0,0 +1,216 @@
+    'order': 'row-major'
+};
+
+var n = scalar2ndarray( 3.0, 'generic', 'row-major' );
```
```suggestion
var n = scalar2ndarray( 3, 'generic', 'row-major' );
```

kgryte replied:
Here, we are representing a count; no need to signal a decimal value.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/README.md:
```diff
@@ -0,0 +1,216 @@
+
+The function accepts the following arguments:
+
+-   **arrays**: array-like object containing an input ndarray and a zero-dimensional ndarray.
```
a zero-dimensional ndarray containing what? You need to be more descriptive here. Telling me that I should provide a zero-dimensional ndarray without telling me what that value is is not particularly insightful.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/README.md:
```diff
@@ -0,0 +1,216 @@
+    'order': 'row-major'
+};
+
+var n = scalar2ndarray( 6.0, 'generic', 'row-major' );
```
```suggestion
var n = scalar2ndarray( 6, 'generic', 'row-major' );
```

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/README.md:
```diff
@@ -0,0 +1,216 @@
+};
+console.log( ndarray2array( x.data, x.shape, x.strides, x.offset, x.order ) );
+
+var n = scalar2ndarray( 5.0, {
```
```suggestion
var n = scalar2ndarray( 5, {
```

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/benchmark/benchmark.1d_rowmajor.js:
```diff
@@ -0,0 +1,138 @@
+		'offset': 0,
+		'order': order
+	};
+	n = scalar2ndarray( 5.0, {
```
```suggestion
	n = scalar2ndarray( 5, {
```

This same change should be made throughout this PR.

kgryte replied:
There seems to be confusion as to when and why we add decimal values. We use decimal values to signal in our documentation when a value can take any real value. That is not the case here. `n` should never be, e.g., `5.5`, as that doesn't make sense. So, leave out the decimal in order to convey that only integer values should be provided.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/docs/repl.txt:
```diff
@@ -0,0 +1,74 @@
+    ----------
+    arrays: ArrayLikeObject<ndarray>
+        Array-like object containing an input ndarray and a 0-dimensional
+        ndarray containing number of elements.
```
```suggestion
        Array-like object containing an input ndarray and a zero-dimensional
        ndarray containing number of elements.
```

And this description needs improvement. Containing what number of elements?

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/docs/repl.txt:
```diff
@@ -0,0 +1,74 @@
+
+    // Using an ndarray...
+    > var x = {{alias:@stdlib/ndarray/ctor}}( dt, xbuf, sh, sx, ox, ord );
+    > var n = {{alias:@stdlib/ndarray/from-scalar}}( 3, { 'dtype': 'generic' });
```
```suggestion
    > var n = {{alias:@stdlib/ndarray/from-scalar}}( 3 );
```

It is okay to use the default dtype here.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/docs/types/index.d.ts:
```diff
@@ -0,0 +1,111 @@
+/**
+* Tests whether at least `n` elements in an ndarray pass a test implemented by a predicate function.
+*
+* @param arrays - array-like object containing an input ndarray and a 0-dimensional ndarray containing the number of elements
```
s/0-dimensional/zero-dimensional/ here and everywhere else.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/docs/types/index.d.ts:
```diff
@@ -0,0 +1,111 @@
+*
+* // Create the input ndarray:
+* var x = ndarray( 'float64', xbuf, shape, sx, ox, 'row-major' );
+* var n = scalar2ndarray( 3.0, { 'dtype': 'generic' } );
```
```suggestion
* var n = scalar2ndarray( 3, { 'dtype': 'generic' } );
```

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/docs/types/index.d.ts:
```diff
@@ -0,0 +1,111 @@
+* var out = someBy( [ x, n ], predicate );
+* // returns true
+*/
+declare function someBy<T = unknown, U = unknown>( arrays: ArrayLike<typedndarray<T>>, predicate: Predicate<T, U>, thisArg?: ThisParameterType<Predicate<T, U>> ): boolean;
```
This isn't typed correctly. It is not necessary that `n` has the same type as the input ndarray. In which case, use

```
arrays: [ typedndarray<T>, typedndarray<number> ]
```

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/examples/index.js:
```diff
@@ -0,0 +1,47 @@
+};
+console.log( ndarray2array( x.data, x.shape, x.strides, x.offset, x.order ) );
+
+var n = scalar2ndarray( 5.0, {
```
```suggestion
var n = scalar2ndarray( 5, {
```

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/0d.js:
```diff
@@ -0,0 +1,84 @@
+* // returns true
+*/
+function some0d( x, n, predicate, thisArg ) {
+	if ( predicate.call( thisArg, x.data[ x.offset ], [], x.ref ) ) {
+		return true;
+	}
+	return false;
```
This is clearly not correct. What if `n > 1`?

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/0d.js:
```diff
@@ -0,0 +1,84 @@
+* @param {IntegerArray} x.strides - stride lengths
+* @param {NonNegativeInteger} x.offset - index offset
+* @param {string} x.order - specifies whether `x` is row-major (C-style) or column-major (Fortran-style)
+* @param {PositiveInteger} n - number of elements
```
s/PositiveInteger/NonNegativeInteger/ here and elsewhere. While `0` and `1` are equivalent, no reason, IMO, to exclude `0` as an "allowed" value.

kgryte replied:
I see that in `utils/some-by` we required a positive integer. I am not sure that was the right decision. `0`, at least to me, seems fine.

kgryte replied:
Actually, thinking about this, if `n = 0`, we'd just add a line in `main.js` to automatically return `true`, as it is trivially true that at least `0` elements satisfy the predicate. In which case, we can leave `PositiveInteger` here and elsewhere.

headlessNode replied:
> if n = 0, we'd just add a line in main.js to automatically return true

Yeah makes sense.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/0d.js:
```diff
@@ -0,0 +1,84 @@
+* @param {PositiveInteger} n - number of elements
+* @param {Function} predicate - predicate function
+* @param {*} thisArg - predicate function execution context
+* @returns {boolean} boolean indicating whether all elements pass a test
```
This description is not correct more generally. This is not `every-by`. Should be

```suggestion
* @returns {boolean} boolean indicating whether at least `n` elements pass a test
```

This applies here and everywhere else in this PR.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/0d_accessors.js:
```diff
@@ -0,0 +1,87 @@
+	if ( predicate.call( thisArg, x.accessors[ 0 ]( x.data, x.offset ), [], x.ref ) ) { // eslint-disable-line max-len
+		return true;
+	}
+	return false;
```
Not correct.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/1d.js:
```diff
@@ -0,0 +1,113 @@
+* @param {PositiveInteger} n - number of elements
+* @param {Function} predicate - predicate function
+* @param {*} thisArg - predicate function execution context
+* @returns {boolean} boolean indicating whether all elements pass a test
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/1d_accessors.js:
```diff
@@ -0,0 +1,120 @@
+* @param {PositiveInteger} n - number of elements
+* @param {Function} predicate - predicate function
+* @param {*} thisArg - predicate function execution context
+* @returns {boolean} boolean indicating whether all elements pass a test
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/1d_accessors.js:
```diff
@@ -0,0 +1,120 @@
+
+	// Iterate over the ndarray dimensions...
+	for ( i0 = 0; i0 < S0; i0++ ) {
+		if ( predicate.call( thisArg, get( xbuf, ix ), [ i0 ], x.ref) ) {
```
```suggestion
		if ( predicate.call( thisArg, get( xbuf, ix ), [ i0 ], x.ref ) ) {
```

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/2d.js:
```diff
@@ -0,0 +1,144 @@
+* @param {PositiveInteger} n - number of elements
+* @param {Function} predicate - predicate function
+* @param {*} thisArg - predicate function execution context
+* @returns {boolean} boolean indicating whether all elements pass a test
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/2d_accessors.js:
```diff
@@ -0,0 +1,151 @@
+* @param {NonNegativeInteger} x.offset - index offset
+* @param {string} x.order - specifies whether `x` is row-major (C-style) or column-major (Fortran-style)
+* @param {Array<Function>} x.accessors - data buffer accessors
+* @param {PositiveInteger} n - number of elements
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/2d_blocked.js:
```diff
@@ -0,0 +1,172 @@
+* @param {IntegerArray} x.strides - stride lengths
+* @param {NonNegativeInteger} x.offset - index offset
+* @param {string} x.order - specifies whether `x` is row-major (C-style) or column-major (Fortran-style)
+* @param {PositiveInteger} n - number of elements
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/1d_accessors.js:
```diff
@@ -0,0 +1,120 @@
+	// Cache accessor:
+	get = x.accessors[ 0 ];
+
+	// Initialize the counter:
```
```suggestion
	// Initialize a counter:
```

Applies here and everywhere else.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/2d_blocked_accessors.js:
```diff
@@ -0,0 +1,179 @@
+	// Cache accessor:
+	get = x.accessors[0];
+
+	// Initialize the counter:
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/3d.js:
```diff
@@ -0,0 +1,154 @@
+* @param {IntegerArray} x.strides - stride lengths
+* @param {NonNegativeInteger} x.offset - index offset
+* @param {string} x.order - specifies whether `x` is row-major (C-style) or column-major (Fortran-style)
+* @param {PositiveInteger} n - number of elements
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/4d_blocked.js:
```diff
@@ -0,0 +1,210 @@
+						for ( i2 = 0; i2 < s2; i2++ ) {
+							for ( i1 = 0; i1 < s1; i1++ ) {
+								for ( i0 = 0; i0 < s0; i0++ ) {
+									if ( predicate.call( thisArg, xbuf[ ix ], take( [ i3, i2, i1, i0 ], idx ), x.ref ) ) { // eslint-disable-line max-len
```
Indices are incorrect.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/4d_blocked_accessors.js:
```diff
@@ -0,0 +1,217 @@
+						for ( i2 = 0; i2 < s2; i2++ ) {
+							for ( i1 = 0; i1 < s1; i1++ ) {
+								for ( i0 = 0; i0 < s0; i0++ ) {
+									if ( predicate.call( thisArg, get( xbuf, ix ), take( [ i3, i2, i1, i0 ], idx ), x.ref ) ) { // eslint-disable-line max-len
```
Indices are incorrect.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/index.js:
```diff
@@ -0,0 +1,72 @@
+*     'order': 'row-major'
+* };
+*
+* var n = scalar2ndarray( 3.0, 'generic', 'row-major' );
```
```suggestion
* var n = scalar2ndarray( 3, 'generic', 'row-major' );
```

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/main.js:
```diff
@@ -0,0 +1,176 @@
+*     -   **offset**: index offset.
+*     -   **order**: specifies whether an ndarray is row-major (C-style) or column major (Fortran-style).
+*
+* @param {ArrayLikeObject<Object>} arrays - array-like object containing an input ndarray and a 0-dimensional ndarray containing number of elements
```
Description.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/main.js:
```diff
@@ -0,0 +1,176 @@
+*     'order': 'row-major'
+* };
+*
+* var n = scalar2ndarray( 3.0, { 'dtype': 'generic' } );
```
```suggestion
* var n = scalar2ndarray( 3, {
*     'dtype': 'generic'
* });
```

Also, in JSDoc examples, we never put on a single line like this. We only do that in REPL help and TypeScript tests. This should be fixed here and anywhere else in this PR.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/main.js:
```diff
@@ -0,0 +1,176 @@
+	// Unpack the ndarray and standardize ndarray meta data:
+	x = ndarray2object( arrays[ 0 ] );
+	n = ndarray2object( arrays[ 1 ] );
+	shx = x.shape;
```
```suggestion

	shx = x.shape;
```

For consistency with https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/ndarray/base/includes/lib/main.js#L254.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/nd.js:
```diff
@@ -0,0 +1,127 @@
+// MAIN //
+
+/**
+* Tests whether all elements in an ndarray pass a test implemented by a predicate function.
```
Description is incorrect.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/nd.js:
```diff
@@ -0,0 +1,127 @@
+* };
+*
+* // Test elements:
+* var out = everynd( x, predicate );
+* // returns true
+*/
+function everynd( x, predicate, thisArg ) {
```
Needs updating.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/nd_accessors.js:
```diff
@@ -0,0 +1,134 @@
+// MAIN //
+
+/**
+* Tests whether all elements in an ndarray pass a test implemented by a predicate function.
```
Needs updating.

kgryte commented on lib/node_modules/@stdlib/ndarray/base/some-by/lib/main.js:
```diff
@@ -0,0 +1,176 @@
+
+	// Resolve the `n` count:
+	n = n.accessors[ 0 ]( n.data, n.offset );
+
```
```suggestion
	if ( n < 1 ) {
		return true;
	}
```

Following up on https://github.com/stdlib-js/stdlib/pull/7087#discussion_r2107818225.

headlessNode committed:
fix: apply review suggestions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
fix: apply suggestions from code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
fix: apply suggestions from code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
feat: add 5d kernels

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
fix: bug in 5d

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
feat: add 6d kernels

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
feat: add 7d kernels

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
feat: add 8d kernels

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
feat: add 9d kernels

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
feat: add 10d kernels

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
feat: add benchmarks & missing imports to main

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
fix: oversight :/

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
bench: avoid constant-time benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
docs: update example

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
test: address lint errors

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte committed:
docs: update examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
docs: update example

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
fix: address empty ndarray case

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
docs: fix typo

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
refactor: consolidate conditions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte committed:
docs: fix missing required argument

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM after clean-up.
